datafusion/optimizer/src/analyzer/mod.rs

```diff
@@ -156,18 +155,21 @@ impl Analyzer {
 /// Do necessary check and fail the invalid plan
 fn check_plan(plan: &LogicalPlan) -> Result<()> {
     plan.apply(&mut |plan: &LogicalPlan| {
-        for expr in plan.expressions().iter() {
+        plan.inspect_expressions(|expr| {
```

Comment:
this is the line that voids a copy

Reply:
A great catch! `expressions()` will invoke clone() inside, we can notice it to reduce some loss